test/solvers/functional/functional_test.py

```diff
@@ -101,20 +109,43 @@ def test_derivative(functional, space):
 
     x = noise_element(functional.domain)
     y = noise_element(functional.domain)
-    epsK = 1e-8
+    epsK = float(np.sqrt(np.finfo(space.dtype).eps))
 
     # Numerical test of gradient
-    assert all_almost_equal(((functional(x + epsK * y) - functional(x)) /
-                            epsK),
-                            y.inner(functional.gradient(x)),
-                            places=PLACES / 2)
+    expected = (functional(x + epsK * y) - functional(x)) / epsK
+    result = y.inner(functional.gradient(x))
+    assert all_almost_equal(expected, result, places=PLACES / 2)
 
     # Check that derivative and gradient is consistent
     assert all_almost_equal(functional.derivative(x)(y),
                             y.inner(functional.gradient(x)),
                             places=PLACES)
 
 
+def test_arithmetic():
+    """Test that all standard arithmetic works."""
+    space = odl.rn(3)
+
+    # Create elements needed for later
+    functional = odl.solvers.L2Norm(space)
```

Comment:
Pick another functional that does not distribute over multiplication, perhaps offset this one.